nilearn/load_confounds/parser.py

```diff
@@ -0,0 +1,290 @@
+"""Flexible method to load confounds generated by fMRIprep.
+
+Authors: load_confounds team
+"""
+import pandas as pd
+from .confounds import (
+    _sanitize_confounds,
+    _confounds_to_df,
+    _prepare_output,
+    MissingConfound,
+)
+from . import components
+
+
+# Global variables listing the admissible types of noise components
+all_confounds = [
+    "motion",
+    "high_pass",
+    "wm_csf",
+    "global",
+    "compcor",
+    "ica_aroma",
+    "scrub",
+    "non_steady_state",
+]
+
+default_parameters = {
+    "strategy": ["motion", "high_pass", "wm_csf"],
+    "motion": "full",
+    "scrub": "full",
+    "fd_thresh": 0.2,
+    "std_dvars_thresh": 3,
+    "wm_csf": "basic",
+    "global_signal": "basic",
+    "compcor": "anat_combined",
+    "n_compcor": "auto",
+    "ica_aroma": "full",
+    "demean": True,
+}
+
+
+def _sanitize_strategy(strategy):
+    """Define the supported denoising strategies."""
+    if not isinstance(strategy, list):
+        raise ValueError("strategy needs to be a list of strings")
+    for conf in strategy:
+        if conf not in all_confounds:
+            raise ValueError(f"{conf} is not a supported type of confounds.")
+    # add non steady state if not present
+    if "non_steady_state" not in strategy:
+        strategy.append("non_steady_state")
+    return strategy
+
+
+def _check_error(missing):
+    """Consolidate a single error message across multiple missing confounds."""
+    if missing["confounds"] or missing["keywords"]:
+        error_msg = (
+            "The following keywords or parameters are missing: "
+            + f" {missing['confounds']}"
+            + f" {missing['keywords']}"
+            + ". You may want to try a different denoising strategy."
+        )
+        raise ValueError(error_msg)
+
+
+def load_confounds(img_files, **kargs):
```

Comment:
I think we should explicitly write the function inputs here to be consistent with the rest of Nilearn's API.